File: Arrays and Hashing/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums):
        prefix = 1
        postfix = 1
        result = []
        for i in range(len(nums)):
            result.append(prefix)
            prefix *= nums[i]
        for i in range(len(nums) - 1, -1, -1):
            result[i] *= postfix
            postfix *= nums[i]
        return result

File: Arrays and Hashing/product-of-array-except-self.js
// time O(n) space O(1)
var productExceptSelf = function (nums) {
  let prefix = 1;
  let postfix = 1;
  let result = Array(nums.length);
  for (let i = 0; i < nums.length; i++) {
    result[i] = prefix;
    prefix *= nums[i];
  }
  for (let i = nums.length - 1; i >= 0; i--) {
    result[i] *= postfix;
    postfix *= nums[i];
  }
  return result;
};
